### DISCUSSION TITLE
when are the platform.sloeber.txt; board.sloeber.txt created?

### DISCUSSION THREAD

**Discussion by PerennialNovice** [Sloeber/arduino-eclipse-plugin#1443] [Q&A] [answered]
I'm looking into creating a NULCEO-L4P5 variant for STM32duino and therefor am modifying boards.txt and some more files.
I now discovered that sloeber creates some workaround files, but I don't know when those files are created, or whether they might be recreated when the original file changes.

Do I have to apply changes to them manually?
Or, if I apply changes to only the .sloeber files, can the changes be overwritten maybe if sloeber automatically recreates these files?

I want to work on the original STM32duino files so if I am successful, I can contribute and make it available for everyone.

**Comment by jantje** [comment, marked as answer]
The *.sloeber.txt files are there for performance reasons and are created/updated when read.
Simplified a read is done when a project is opened (for the first time?) or when project properties->sloeber apply is done.

### The read procedure is as follows

1. The first line of ToRead.sloeber.txt is read and checked against a hard coded string in sloeber (this is to cope with new Sloeber versions that use a different workaround)
2. If the first line is different from the hardcoded line the ToRead.sloeber.txt file is deleted.
3. If no ToRead.sloeber.txt is found the workaround file is created based on the ToRead.txt.
4. The ToRead.sloeber.txt is read

### My advice for platform developers in regards to modifying *.txt *.sloeber.txt files
I advice to focus on making your changes to the root files (boards.txt platform.txt programmers.txt) and delete the sloeber workaround versions after you modified a root file.
My reasons are

- you can work in the documented versions in the files and not in the undocumented sloeber versions.
- The *.sloeber.txt are derived from the *.txt so you will have to modify the *.txt anyway.
- When a *.sloeber.txt is modified; your setup is in a quantum superposition state; which may cause confusion 😉 .

If the workaround implementation that the sloeber version you are using, doesn't work for you; make an issue explaining the problem and _**make it very clear you are doing your own platform development.**_

**If you want to test your changes.** 

1. Save you *.txt files
2. Delete the corresponding *.sloeber.txt files
3. Goto project properties->sloeber and select apply/close.
4. optionally you can verify that your changes have been read by checking the existence of the *.sloeber.txt file.
5. Do your test

Your questions
>Do I have to apply changes to them manually?

No see above

>Or, if I apply changes to only the .sloeber files, can the changes be overwritten maybe if sloeber automatically recreates these files?

They will be overwritten when sloeber is upgraded to a new version that contains a different workaround implementation.
They may be overwritten in other cases (probably not but architecturally speaking these files are there for performance reasons so there is no guarantee about there lifecycle from a user point of view)

>I want to work on the original STM32duino files so if I am successful, I can contribute and make it available for everyone.

Great. If you work as described above that seems a very legit way to work to me.

**Reply by PerennialNovice** [reply]
Thank you @jantje, for this very abundant and helpful answer :)
